### PR TITLE
openblas: add rpi support

### PR DESCRIPTION
**Pull request by TSAnsari**
The recipe was using the incorrect ARM target for
raspberry pi. This resulted in failing cross compilation.
This corrects the recipe to use ARMV7 as target. It also
adds the following cmpiler options. These options enable
the use of FPU for floating point calculations instead of
software emulation.

-mfpu=neon-vfpv4
-mfloat-abi=hard

Signed-off-by: Tariq Ansari <tansari@luxoft.com>

### CODE REVIEW

File: recipes-alexa/openblas/openblas_0.3.5.bb [hunk 1]
```diff
@@ -28,6 +28,7 @@ def map_arch(a, d):
         elif re.match('x86_64$', a): return 'ATOM'
         elif re.match('aarch32$', a): return 'CORTEXA9'
         elif re.match('aarch64$', a): return 'ARMV8'
+        elif re.match('arm$', a): return 'ARMV7'
```
Comment (sashko) [inline recipes-alexa/openblas/openblas_0.3.5.bb]: where is this used? `Cortex-A53` is `armv8`

File: recipes-alexa/openblas/openblas_0.3.5.bb [hunk 2]
```diff
@@ -36,11 +37,17 @@ def map_bits(a, d):
         elif re.match('x86_64$', a): return 64
         elif re.match('aarch32$', a): return 32
         elif re.match('aarch64$', a): return 64
+        elif re.match('arm$', a): return 32
```
Comment (sashko) [inline recipes-alexa/openblas/openblas_0.3.5.bb]: is `armv8` set as well? if so, use it

File: recipes-alexa/openblas/openblas_0.3.5.bb (hunk 1 above)
Comment (TSAnsari) [reply recipes-alexa/openblas/openblas_0.3.5.bb]: I think armv7 is set in raspberry pi bsp layer, so we have to use ARMV7. All rpi builds in PELUX are cross compiled for armv7.

File: recipes-alexa/openblas/openblas_0.3.5.bb (hunk 2 above)
Comment (TSAnsari) [reply recipes-alexa/openblas/openblas_0.3.5.bb]: no armv8 is not set.

Comment (TSAnsari) [reply recipes-alexa/openblas/openblas_0.3.5.bb]: The following are set for rpi build and can be seen when bitbake starts

TUNE_FEATURES        = "arm armv7ve vfp thumb neon vfpv4 callconvention-hard cortexa7"

Comment (TSAnsari) [reply recipes-alexa/openblas/openblas_0.3.5.bb]: So we are always compiling for armv7. Probably to have backwards compatibility.

**Comment by thenor** [comment]
> The recipe was using the incorrect ARM target for
> raspberry pi. This resulted in failing cross compilation.
> This corrects the recipe to use ARMV7 as target. It also
> adds the following cmpiler options:
> 
> -mfpu=neon-vfpv4
> -mfloat-abi=hard
>
>These options enable the use of FPU for floating point calculations instead of
> software emulation.
> 
> Signed-off-by: Tariq Ansari [tansari@luxoft.com](mailto:tansari@luxoft.com)

I would like a small restructure of the commit message, for clarity.

**Comment by TSAnsari** [comment]
> > The recipe was using the incorrect ARM target for
> > raspberry pi. This resulted in failing cross compilation.
> > This corrects the recipe to use ARMV7 as target. It also
> > adds the following compiler options:
> > -mfpu=neon-vfpv4
> > -mfloat-abi=hard
> > These options enable the use of FPU for floating point calculations instead of
> > software emulation.
> > Signed-off-by: Tariq Ansari [tansari@luxoft.com](mailto:tansari@luxoft.com)
> 
> I would like a small restructure of the commit message, for clarity. There was also a typo ("compiler").

Sure, will do that

**Pull request merged**